NomaiGrandPrix/NomaiGrandPrix.cs

```diff
@@ -191,8 +208,13 @@ private void SpeedRunButton_OnClick()
 
         private void ResetRunButton_OnClick()
         {
-            SpeedrunState.SpawnPoint = GetRandomSpawnConfig(config => { return config.shouldSpawn; });
-            SpeedrunState.GoalPoint = GetRandomSpawnConfig(config => { return config.shouldGoal; });
+            //TODO: figure out a good place to consolidate this logic since it's duplicated 4 times.
```

Comment:
It looks like it's only being used in this class; would a private function encapsulating it be sufficient?